.buildkite/scripts/steps/create_dra.sh

```diff
@@ -0,0 +1,27 @@
+#!/bin/bash
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License
+# 2.0 and the following additional limitation. Functionality enabled by the
+# files subject to the Elastic License 2.0 may only be used in production when
+# invoked by an Elasticsearch process with a license key installed that permits
+# use of machine learning features. You may not use this file except in
+# compliance with the Elastic License 2.0 and the foregoing additional
+# limitation.
+
+# This post-processing step of ML C++ CI does the following:
+#
+# 1. Download the platform-specific artifacts built by the first phase
+#    of the ML CI job.
+# 2. Combine the platform-specific artifacts into an all-platforms bundle,
+#    as used by the Elasticsearch build.
+
+rm -rf build/distributions
+
+# Default to a snapshot build
+if [ -z "$BUILD_SNAPSHOT" ] ; then
+    BUILD_SNAPSHOT=true
+fi
+
+# Download from S3, combine, and upload to BuildKite's artifact store.
```

Comment:
For BuildKite we shouldn't download the platform-specific artifacts from S3. Instead we should use BuildKite's artifact propagation mechanism to pass them into this final step.

Under Jenkins we used S3 for two purposes:

1. As an intermediate store of platform-specific artifacts from the platform-specific builds
2. As a permanent store of the latest artifacts built for each branch

In BuildKite we only need the second of these, because BuildKite itself provides a better solution for the first requirement.

We can have a step at the very end of the branch builds (_only_ branch builds) that:

- Gets all the platform-specific artifacts stored by previous steps
- Combines them into the all-platforms artifacts
- Gets AWS credentials
- Uploads all artifacts to S3
- Runs the DRA Docker container to do its stuff